aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -655,6 +656,16 @@ std::tuple<Tensor, Tensor, Tensor, Tensor, Tensor> _cudnn_rnn(
     const Tensor& fn_dropout_state
     ) {
 
+  // check tensor device types
+  auto to_check_tensors = std::vector<TensorArg>();
+  to_check_tensors.emplace_back(TensorArg(input_r, "input", 1));
+  to_check_tensors.emplace_back(TensorArg(hx, "hx", 2));
+  to_check_tensors.emplace_back(TensorArg(cx, "cx", 3));
+  for (int i = 4; i < weight.size(); i++) {
+    to_check_tensors.emplace_back(TensorArg(weight[i], ("weight_" + std::to_string(i)).c_str(), i));
```

Comment:
Those are not the real names of those weights, so it will only end up confusing. They're usually called sth like `weight_ih_l1_rev`. We don't want to format strings like this every time this is called, so it's really better to just say that "weights were found on device X, but were expected to be on device Y", without getting into specifics.

Also, it would be good if we could avoid allocating the vector unnecessarily. Your previous function did a pretty good job at that.

Reply:
Thanks for pointing this out! I will bring back the previous `check_device` function then.